### ISSUE TITLE
Swap OK/Return buttons

### ISSUE THREAD

**Issue by jesus2099** [Aloshi/EmulationStation#627]
By default, EmulationStation happnes to be:

- A for Cancel
- B for Confirm

It took me times to get used to it but I still mistake quite regularly.
Coud there be an **option** for swapping this to: 

- A for Confirm
- B for Cancel

Not just swapping my controller settings because then, the problem would just move from ES to all games. :)

**Comment by pumrum** [comment]
I agree that the default A/B for cancel/confirm in EmulationStation is contrary to what most gamepad users are accustomed to (especially those using playstation controllers). While we wait for word on a permanent fix, you can edit the following file to fix only EmulationStation without affecting the rest of your emulators/roms. This assumes you're using RetroPie on a Raspberry Pi, on other systems YMMV:

::swap A/B buttons in emulation station
/opt/retropie/configs/all/emulationstation/es_input.cfg
<input name="a" type="button" id="1" value="1"/>
<input name="b" type="button" id="2" value="1"/>

RetroArch includes a specific "Menu Swap Ok & Cancel Buttons" option that has the same effect on those interfaces.

**Comment by jesus2099** [comment]
Retroarch and all games have correct setting except EmulationStation which has B (down) to confirm and A (right) to cancel, so I would only want to change this.

For the record, B/**A** (down/**right**) →cancel/**ok** is the usual Nintendo setting and **Japanese** PlayStation setting.

I use recalbox so this file was found in `\\Recalbox\share\system\.emulationstation\es_input.cfg`.
I changed only my gamepad `SFC30` (there are many entries there) and changed only this (swapped `name="a"` ↔ `name="b"`).

The problem is that it has swapped everything, not only the Emulation Station menu. :/

There is a `\\Recalbox\share\system\configs\retroarch\retroarchcustom.cfg` file which contains `input_menu_toggle_btn = 1` so I tried unswapping the buttons (`input_menu_toggle_btn = 0`) but it didn’t change a thing and anyway I doubt it would have restored correct game button layout.

So I have swapped back `name="b"` ↔ `name="a"` in `es_input.cfg` until some other tip is found. :)

Thanks very much, @pumrum, for letting me know about these `cfg` file, I’ll explore them to find maybe a solution !

**Comment by erwan** [comment]
What's the status on that, any reliable solution found?

**Comment by adventureFAN** [comment]
Hope this option will add... because I like batocera very much but this issue (for me) confused me every time >_>

**Comment by NextBlaubeere** [comment]
This issue especially matters for people who do play NES/SNES games in addition to other platforms.

**Playing most platformers:** inside button is OK and outside button is CANCEL.
**Playing NES games:** inside button is CANCEL and outside button is OK. 

Is there a way, to set _"menu_swap_ok_cancel_buttons"_ per controller? So I would be able to set this _"true"_ for my NES controller and not the others. With this mechanism in place it wouldn't feel weird when I go from a NES game into the menu, because ingame (NES game) and the menu (retroarch) would be the same.
